phoebe/dependencies/ndpolator/ndpolator.py

```diff
@@ -13,13 +39,13 @@ def __init__(self, axes, grid, impute=False):
 
         if impute:
             raise NotImplementedError('consider if this is worth implementing here.')
-        
+
         self.indices = np.argwhere(~np.isnan(grid[...,0]))
         non_nan_vertices = np.array([ [self.axes[i][self.indices[k][i]] for i in range(len(self.axes))] for k in range(len(self.indices))])
         self.nntree = cKDTree(non_nan_vertices, copy_data=True)
         ranges = (range(len(ax)-1) for ax in axes)
         slices = [tuple([slice(elem, elem+2) for elem in x]) for x in product(*ranges)]
-        self.ics = np.array([s.start for slc in slices for s in slc if ~np.any(np.isnan(grid[slc]))], dtype=int).reshape(-1, len(axes))
+        self.ics = np.array([s.start for slc in slices for s in slc if ~np.any(np.isnan(grid[slc]))], dtype=int).reshape(-1, len(axes))  # bottleneck
```

Comment:
how much of a bottleneck?  Should we open a ticket so that we remember to get back to this someday?

Reply:
this is being ported into C right now.